test/hotspot/jtreg/compiler/vectorization/TestPopulateIndex.java

```diff
@@ -112,4 +112,22 @@ public void checkResultExprWithIndex2() {
             }
         }
     }
+
+    @Test
+    @IR(counts = {IRNode.POPULATE_INDEX, "= 0"}) // disabled by sum-under-mask optimization.
+    public void exprWithIndex3() {
+        for (int i = 0; i < count; i++) {
+            dst[i] = src[i] * (i & 7);
+        }
+        checkResultExprWithIndex3();
+    }
```

Comment:
I'm afraid this will now vectorize on some platforms and not on others:
If the number of elements in the vector is <=8 then your optimization probably does not kick in, but with larger vectors, we would get `(i + 8) & 7` folded to `i & 7`, and only at this point does it not vectorize as expected...

In this case, I'm ok with just removing the IR rule. Splitting the IR rule is going to be a pain.

But let's leave a comment here that we might want to fix this in a future RFE.

Reply:
Done.